lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -38,6 +38,8 @@ def type_for(column_name)
               "::DateTime"
             when ActiveRecord::AttributeMethods::TimeZoneConversion::TimeZoneConverter
               "::ActiveSupport::TimeWithZone"
+            when ActiveRecord::Enum::EnumType
+              "::String"
```

Comment:
Thanks for the review! Good catch on the string versus integer column. I covered the casting/coercion question a bit in my PR description:

> I added support for ActiveRecord::Enum::EnumType in ActiveRecordColumnTypeHelper. I considered including Symbol in allowed setter types as that is common in our codebase, but I left it out since I noticed no other types seem to cover common castable inputs. In some other cases the automatic casting results can be surprising in my opinion, so I can see arguments behind favoring explicit casting and keeping the supported setter types narrow.

I'm happy to update this to cover all the acceptable types for enums. Is that something the team agrees on?

Reply:
Yeah good catch, let's broaden the type 👍

Reply:
Thanks for the details @paracycle. I updated the setter type to be conditional on subtype as you suggested.